### ISSUE TITLE
LargeHeader stays small after pop/goBack/swipe gesture on iOS 14+

### ISSUE THREAD

**Issue by Hirbod** [software-mansion/react-native-screens#649]
Hi,

I just upgraded to Expo SDK 39 and before of that, I had never an issue navigating to a child screen (from LargeHeader to small header). When I go back now, it stays smalls instead of growing back to large.

What has changed and is this something known?

**Comment by WoLewicki** [comment]
Yeah, I can spot problems with this. It looks like some kind of bug because it doesn't happen in the `Settings` of your phone. Am I right?

**Comment by Hirbod** [comment]
@WoLewicki you are right, works great everywhere else in the system.

**Comment by ArekChr** [comment]
The same here 👍

**Comment by WoLewicki** [comment]
Does this behavior appear only while using `ScrollView`?

**Comment by Hirbod** [comment]
For me, it happens when using `ScrollView` and `FlatList`

**Comment by julian-baumann** [comment]
Maybe this is related: [iOS 14 Large title collapse when app run firstly](https://developer.apple.com/forums/thread/660745).

I experience this issue too, when I start the app or scroll to the top, navigate and go back on iOS 14.

**Comment by WoLewicki** [comment]
Can you check if #670 fixes the issue and does not introduce any new issues?

**Comment by WoLewicki** [comment]
Looks like it works @grahammendick :tada:. Are you going to submit a PR with this in the `react-native` repository?

**Comment by grahammendick** [comment]
@WoLewicki I'm glad it works for you too!
I don't really like submitting PRs in the react-native repo. It takes me a while to set everything up to run locally and then they never look at the PR anyway. I'm happy for someone on this issue to submit the PR instead if they want

**Comment by grahammendick** [comment]
@Hirbod You're right, but the react-native repo is just one example of the problems you have if you're an unknown name in React. I've found it impossible to get my work looked at. If you're  not a name then you're not getting in

**Comment by ArekChr** [comment]
@WoLewicki can someone from your team create PR to react-native repository?

**Comment by WoLewicki** [comment]
It would need more debugging, submitting a PR with changes to a component used by almost every developer must have a very strong explanation and must ensure that no right behavior will be broken after it.

**Comment by thomaswang** [comment]
I have large headers enabled using Expo SDK 39, but when the app is first opened, the large header is collapsed. It happens on iOS 14 and Simulator. @WoLewicki I added the patch https://github.com/software-mansion/react-native-screens/pull/670 but it doesn't change anything.

**Comment by DasArthur** [comment]
For me, this quick hack worked.
If you are using Flatlist or ScrollView even then before <FlatList> add <View></View> and wrap it all in the <> </>

Example code:

```javascript
return(
  <>
      <View></View>
      <Flatlist> 
          {/*your content goes here */}
       </Flatlist>
  </>
)
```

**Comment by WoLewicki** [comment]
@DasArthur in my test cases, it looks like there is some race condition and it sometimes loads initially a large header and sometimes a small one. Can you check this solution @ArekChr @grahammendick ?

**Comment by grahammendick** [comment]
I remember trying a variety of contentOffset and contentInset and repeatedly reloading  - sometimes it worked and sometimes it didn't. My guess is that the times it worked is when the [reactSetFrame was called after it was added to the window](https://github.com/software-mansion/react-native-screens/issues/649#issuecomment-712199895) - @DasArthur Could you check if the timing of the reactSetFrame changes when it works and when it doesn't? Or does it always work for you every time you reload?

**Comment by ArekChr** [comment]
I don't have maintainVisibleContentPosition prop, @DasArthur what version of RN are you using?

**Comment by grahammendick** [comment]
The [workaround](https://github.com/software-mansion/react-native-screens/issues/645#issuecomment-710033686) of @alexanderstrom seems to be the only reliable one. This is where you conditionally render the ScrollView after a delay (useEffect with a timeout of 0). I think it works for the same reason that [my fix](https://github.com/software-mansion/react-native-screens/issues/649#issuecomment-712199895) works  - the delayed render means that `didMoveToWindow` runs before `reactSetFrame`. It runs as soon as it’s added as a subview because the parent is already in the window.

**Comment by Hirbod** [comment]
@grahammendick still feels so hacky :/

**Comment by grahammendick** [comment]
@Hirbod I think another option is you could use patch-package to apply [the fix to RCTScrollContentView](https://github.com/software-mansion/react-native-screens/issues/649#issuecomment-712199895)

**Comment by Hirbod** [comment]
@grahammendick unfortunately not an option for me, because I'm still on Expo and I don't want to eject. Since I only have 3 LargeTitle Screens, I'll try the delay render useEffect hack.

**Comment by grahammendick** [comment]
@Hirbod You can see that [I'm using it in the zoom example app](https://github.com/grahammendick/navigation/blob/37096f0d605a71eb1b97db2279da9aad07930b0a/NavigationReactNative/sample/zoom/Grid.js#L45) in my Navigation router

**Comment by Hirbod** [comment]
Weird, for me it was simply enough to add this:

```tsx
        <>
            <View></View>
            <FlatList>...
        </>
```

So just empty brackets surrounding and an empty view before my Scrollview.

**Comment by Hirbod** [comment]
Ohhh... ok, my hack "works" for the back button, but it won't collapse when I scroll now. Dang it!

**Comment by Hirbod** [comment]
None of those workarounds working for me, even delaying the Scrollview. I don't have any INITIAL issues with the LargeHeader, it is working for me always, its just when navigating to a stack-screen and returning. The header stays small for me after I pop. Only my mentioned "workaround" fixes this, but will break collapsing on scrolling :/

**Comment by grahammendick** [comment]
@Hirbod Ok, [my fix](https://github.com/software-mansion/react-native-screens/issues/649#issuecomment-712199895) probably doesn't work for you either, then. Can you temporarily eject from expo, apply the fix and let me know if it works, please?

**Comment by happyfloat** [comment]
@grahammendick I tested it on iOS so far and it seems to work well so far! Thank you.

Here the patch `react-native+0.63.3.patch` for [patch-package](https://github.com/ds300/patch-package#readme):

```js
diff --git a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
index cf6a0b1..4d187db 100644
--- a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
+++ b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
@@ -14,13 +14,20 @@
 
 @implementation RCTScrollContentView
 
+- (void)didMoveToWindow
+{
+    [super didMoveToWindow];
+    RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
+    [scrollView updateContentOffsetIfNeeded];
+}
+
 - (void)reactSetFrame:(CGRect)frame
 {
   [super reactSetFrame:frame];
 
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
 
-  if (!scrollView) {
+  if (!scrollView || !self.window) {
     return;
   }
```

**Comment by hsavit1** [comment]
I used the useIsFocused() hook from react navigation

**Comment by Hirbod** [comment]
For doing what?

**Comment by dhanushm360** [comment]
Is there any update on this?

**Comment by chinieer** [comment]
same issus from expo users

**Comment by fdecampredon** [comment]
Just for record I applied the @grahammendick patch, it works for initial rendering, but not for back

**Comment by Hirbod** [comment]
@grahammendick I can also confirm that I've made a temp eject and your applied patch did not fix the grow to large after a "pop / back". Still stays small.

**Comment by grahammendick** [comment]
@Hirbod @fdecampredon I applied the patch to the [zoom sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/zoom) in my Navigation router and it works for both initial rendering and after a pop/back. Give the sample a try and it might give you a clue to what's different about React Navigation?

**Comment by pouyarezvani** [comment]
@grahammendick Where do I paste [your fix](https://github.com/software-mansion/react-native-screens/issues/645#issuecomment-712199841) I can't seem to find a file called `RCTScrollContentView` within my react native project.

**Comment by pouyarezvani** [comment]
@Hirbod & @WoLewicki Any updates/Luck on this issue? My 4 bottom navigation tabs all render a large header, and each page has a scrollView, but on initial render the large header is small and only is large by dragging the screen down.

**Comment by sallar** [comment]
@pouyarezvani you have to use `react-native-patch` library for this

**Comment by pouyarezvani** [comment]
@sallar Has there not been a fix for this issue on the live/working/documented branch? 
And so I have to `patch-package` to fix the issue?

**Comment by nguyenbathanh** [comment]
Doe anyone find out how to fix this issue without touching native code as @grahammendick 's fix?

**Comment by nguyenbathanh** [comment]
I just ejected project to Bare Workflow and apply @grahammendick 's patch, it's not working.

**Comment by trajano** [comment]
> For me, it happens when using `ScrollView` and `FlatList`

I'd like to add `SectionList` so it's likely a `<VirtualizedList>` issue.  I just referenced this on Stack Overflow I didn't know about this bug. https://stackoverflow.com/questions/68477361/how-do-i-make-headerlargetitle-appear-as-large-title-when-the-component-is-a-fla

**Comment by WoLewicki** [comment]
@trajano There is a PR implementing this, but it is a workaround of the issue rather than solution, so it can introduce other, not easy to track problems, so it probably won't be merged: https://github.com/software-mansion/react-native-screens/pull/670

**Comment by aryanm5** [comment]
> @grahammendick I tested it on iOS so far and it seems to work well! Thank you.
> 
> Here the patch `react-native+0.63.3.patch` for [patch-package](https://github.com/ds300/patch-package#readme):
> 
> ```js
> diff --git a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
> index cf6a0b1..4d187db 100644
> --- a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
> +++ b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
> @@ -14,13 +14,20 @@
>  
>  @implementation RCTScrollContentView
>  
> +- (void)didMoveToWindow
> +{
> +    [super didMoveToWindow];
> +    RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
> +    [scrollView updateContentOffsetIfNeeded];
> +}
> +
>  - (void)reactSetFrame:(CGRect)frame
>  {
>    [super reactSetFrame:frame];
>  
>    RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
>  
> -  if (!scrollView) {
> +  if (!scrollView || !self.window) {
>      return;
>    }
> ```

This worked for me in 0.63 and 0.64, but after upgrading to 0.65 I am getting a build time error due to this patch:
`ARC semantic issue`
`No visible @interface for 'RCTScrollView' declares the selector 'updateContentOffsetIfNeeded'`
If I disable the patch, the app builds but the big header problem comes back (the header starts in collapsed mode). Is there a solution for 0.65?

**Comment by andreacassani** [comment]
@aryanm5 updateContentOffsetIfNeeded changed to updateContentSizeIfNeeded in 0.65, below the patch `react-native+0.65.1.patch`:

```Objective-C
diff --git a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
index 8006540..a2080ff 100644
--- a/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
+++ b/node_modules/react-native/React/Views/ScrollView/RCTScrollContentView.m
@@ -14,13 +14,20 @@
 
 @implementation RCTScrollContentView
 
+- (void)didMoveToWindow
+{
+  [super didMoveToWindow];
+  RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
+  [scrollView updateContentSizeIfNeeded];
+}
+
 - (void)reactSetFrame:(CGRect)frame
 {
   [super reactSetFrame:frame];
 
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
 
-  if (!scrollView) {
+  if (!scrollView || !self.window) {
     return;
   }
```

**Comment by grahammendick** [comment]
@aryanm5 Looks like [react native has renamed updateContentOffsetIfNeeded](https://github.com/facebook/react-native/pull/30647) to `updateContentSizeIfNeeded`.

Are you interested in creating this patch as a pull request on react native so that this problem doesn't happen again? Looks like react native are looking at pull requests again so might be worth giving it a go.

**Comment by aryanm5** [comment]
> @aryanm5 Looks like [react native has renamed updateContentOffsetIfNeeded](https://github.com/facebook/react-native/pull/30647) to `updateContentSizeIfNeeded`.
> 
> Are you interested in creating this patch as a pull request on react native so that this problem doesn't happen again? Looks like react native are looking at pull requests again so might be worth giving it a go.

Sure. I'll give it a go and post updates here.

**Comment by aryanm5** [comment]
@grahammendick [Here's the pull request](https://github.com/facebook/react-native/pull/32135)

**Comment by grahammendick** [comment]
@aryanm5 Thanks for stepping up. I really appreciate it

**Comment by grahammendick** [comment]
Hey @aryanm5, 

You can point Phillip at my [zoom sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/zoom). It’s a reproduction of the large title problem patched with the fix in the PR. If he removes the `!self.window` check then the fix won’t work anymore (the large title starts small). I don’t know why it doesn’t work. I just discovered the fix by playing around.

From my testing, we don’t need the fix on iOS 15. @aryanm5, can you check large titles work for you on iOS 15 without the fix, please?

Yes, a version check for iOS 14 makes sense.

**Comment by WoLewicki** [comment]
@Hirbod do you have knowledge if such behavior happens in other libraries doing kind of the same thing, like `react-native-navigation` or @grahammendick Navigation router ? And if so, did they come up with a solution?

**Comment by Hirbod** [comment]
@WoLewicki unfortunately not. I never tried another library. I can only tell you that it doesn’t happen in other system apps or any app I am using.

**Comment by WoLewicki** [comment]
It is probably another issue of `react-native`'s `ScrollView` and large header. Could you post here a quick reproduction of the newest problem?

**Comment by grahammendick** [comment]
@Hirbod @WoLewicki The Navigation router doesn't have any problem with large headers on iOS 15. You could run my [zoom sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/zoom) and compare view hierarchies?

**Comment by nandorojo** [comment]
> iOS 15 fixed the bug I've reported initally but introduced a new one. The large header is now snapping to a small header for me as soon as I scroll, looking super weird. (Expo SDK 43, iOS 15, iPhone 13 Pro Max)
> 
> https://streamable.com/x57lig

I am having the exact same issue as @Hirbod on iOS 15; glad I found this thread. I'm using `react-native-screens@3.9.0` with `@react-navigation/native-stack@6.2.2` and `react-native@0.64.3`. I'll try to make a repro.

**Comment by nandorojo** [comment]
> It is probably another issue of `react-native`'s `ScrollView` and large header. Could you post here a quick reproduction of the newest problem?

@WoLewicki Here is a reproduction:

## Video



| iOS 14 (works) | iOS 15 |
| ------ | ------ |
| <video src="https://user-images.githubusercontent.com/13172299/143783461-ef03f422-ff88-4398-967d-bf33b27e96a5.mp4" /> | <video src="https://user-images.githubusercontent.com/13172299/143783481-5f15f563-9f90-4d30-a31d-4d99e792b538.MP4" /> |

## Code

Repo here: https://github.com/nandorojo/large-title-bug

<details>
  <summary>Click to view code</summary>

```jsx
import React from "react";
import { ScrollView, StyleSheet, Text, View } from "react-native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { NavigationContainer } from "@react-navigation/native";

const { Navigator, Screen } = createNativeStackNavigator();

function Home() {
  return (
    <ScrollView>
      {new Array(40).fill(0).map((_, i) => (
        <View style={[styles.item, i % 2 && styles.odd]}>{i}</View>
      ))}
    </ScrollView>
  );
}

export default function App() {
  return (
    <NavigationContainer>
      <Navigator>
        <Screen
          name="Home"
          component={Home}
          options={{
            headerLargeTitle: true,
          }}
        />
      </Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  item: {
    height: 100,
    backgroundColor: "#41b87a",
  },
  odd: {
    backgroundColor: "#333333",
  },
});
```
</details>

**Comment by Hirbod** [comment]
@nandorojo thanks for providing a repro for the issue. My desk was overloaded and I haven’t managed to create one!

**Comment by WoLewicki** [comment]
@nandorojo the reproduction seems to be the same issue as the old ones with `ScrollView` and was resolved by this: https://github.com/software-mansion/react-native-screens/issues/320#issuecomment-670469660. I checked the repro and it works correctly when applying those props. Can you check if it fixes your issue? And if so, is there something more to be done about it?

**Comment by Hirbod** [comment]
I don’t like the fact we need to set special props in order to make it work. It wasn’t like this till iOS 13, broken in iOS 14 and now semi working on iOS 15 (but working under special circumstances when changing header and Scrollview).

Is there a way to make the DX better? I mean there is also no documentation about this.

**Comment by WoLewicki** [comment]
I am afraid these options are the defaults used by native apps on `iOS` and are required for correct behavior of `ScrollView` integrated with large header.

**Comment by sallar** [comment]
@WoLewicki Then I guess this issue can be closed right? Since it's not actually a bug?

**Comment by WoLewicki** [comment]
@sallar yeah, I also think there is not much to be added from the `react-native-screens` side. I will close it then, feel free to comment here if something is wrong and we can always reopen this.

**Comment by toriqo** [comment]
> For me, this quick hack worked.
> 
> ```js
> return(
> 
>       <FlatList
>            contentOffset={{ x: 0, y: -1}}
>                 contentInsetAdjustmentBehavior={"always"}
>                 maintainVisibleContentPosition={
>                     {
> 
>                         autoscrollToTopThreshold: -1,
>                         minIndexForVisible: -1
>                     }
>                 }
>        > 
>           {/*your content goes here */}
>        </FlatList>
> )
> ```

actually, for me, just adding `contentInsetAdjustmentBehavior="automatic"` to my flat list did the trick

**Comment by trajano** [comment]
@branaust you may have to bite the bullet a bit on this one and create your own header/navigator yourself.  I had to do that on my project and spent quite a bit of time trying to get the nuances of the animation correct.  However, the header behaviour is predictable after that.  I am just revisiting the idea again and posted to SO https://stackoverflow.com/questions/74957469/can-a-custom-header-in-react-navigation-receive-the-scrollview-ref-in-a-createst to see if there's a saner approach than what I did.

**Comment by trajano** [comment]
@branaust what if you did 
```
const setRenderScrollViewToTrue = useCallback(()=>setRenderScrollView(true));
...
<Component onLayout={setRenderScrollViewToTrue} />
```

Would you eliminate the "timeout"

**Comment by dioncodes** [comment]
> Unfortunately, this is still an issue for me. None of the solutions in this thread have fixed this behavior. The problem seems to stem from hiding the parent navigators header and then navigating to the child screen with largeHeaderTitle set to true.

Probably not the answer you are looking for and an unpopular opinion, but if you're early in the development I can only recommend not to use React Native 😄. Use Ionic or native development instead... RN comes with tons of dependencies and people tend to use a lot of libraries. Whenever an update comes out either for the OS or React Native pretty much everything breaks. I previously tried realizing a large app project with React Native and more time was spent on maintaining, forking, manually patching libraries and investigating than the usage of RN saved in time... And nowadays in most cases web apps like Ionic feel quite native as well and save A LOT of time in development. RN is just something in between (web and native) but not doing any of both really well.

**Comment by trajano** [comment]
@dioncodes I somewhat agree with you, but I also came from the hell of Ionic v2->v3->v4->aborted v5  upgrades which require me to restructure my entire code base to get things working.  Working on React Native is actually less painful once you have control of the dependencies (which takes a bit of time but I think more fruitful), but Ionic upgrades require project structure changes which means you'd have to know what everyone has done and migrate over.

NativeScript would've been a good option if [HMR with Vue worked](https://github.com/NativeScript/NativeScript/issues/9507).

[Vue Native](https://github.com/GeekyAnts/vue-native-core) has been deprecated.  Their team has focus has moved to NativeBase.

The primary selling point for React Native for me is still `Expo Go` and the snacks which allows me to quickly prototype UIs and show without huge drama.  Without Expo, React-Native is too much of a pain to work with alone.

As far as managing dependencies learning from experience, whenever we use any 3rd party dependency, we have to think of how to get rid of it asap or merge their code base into our own using yarn workspaces (haven't gotten that far myself, what I've done is clone and push to our internal npmjs with updated package.json once we verify it works correctly in the current versions of React(native).  The issue also appears with Ionic though not as much because Angular tends to control everything for you much like Vue.

**Comment by dioncodes** [comment]
@trajano Yeah, Expo and the previews are definitely a good point with RN. However, you also get these with Ionic or web based frameworks, because you can already preview everything in your browser including HMR... The upgrade paths for Ionic weren't that well either in history, ngl. But it's less of an issue to stick to a previous Ionic version because most is still compatible to newer OS as there are less native components. I have maintained some apps that are running in latest OS versions on a code base that is 3 years old without any migrations so far.

One more reason why I stopped working with RN was that there were a few bugs with native libraries (including this react-native-screens) that no solutions existed for yet. Issues and comments were all over Github (including the official RN repo, just take a look at how many unsolved issues there are). And fixing them on your own would be a lot of work and require knowledge of the libraries or RN itself including its ancient Objective-C code. Maybe it was bad luck but during my larger RN project I stumbled in so many issues that were so time intense to solve, that it would have been much quicker to build a completely native app.

Anyways I'm also not a big fan of React anymore, especially since v3 Vue became so much better and that's already another reason for me not work with RN. And Vue Native isn't a reasonable option either (as you already mentioned). I'll just stick to Vue Ionic if Android is necessary and Swift + SwiftUI otherwise. In the end everything other from completely native development is a compromise. However native Android development is such a PITA that at least for Android some web/hybrid solution is necessary (if the platform needs to be supported at all) and dedicated/separate apps for iOS and Android require a higher budget...

**Comment by dvcrn** [comment]
2 years old issue, but since I'm dealing with this right now as well, some observations on my specific case

Code:
```ts
  return (
    <Stack.Navigator
      screenOptions={{
        headerLargeTitle: true,
        headerShown: true,
        headerShadowVisible: false,
        headerLargeTitleShadowVisible: false,
        headerStyle: {
          backgroundColor: '#F7F6FB',
        },
      }}
    >
      <Stack.Screen
        options={{
          headerLeft: () => (isFetching && <ActivityIndicator />) || <></>,
          headerRight: () => (
            <Button onPress={() => navigation.navigate('Settings')} title="Settings" color="#000" />
          ),
        }}
        name="My Header"
        component={HomeScreen}
      />
    </Stack.Navigator>
  );
```

where HomeScreen is returning a `FlatList` inside a `View`

```ts
        <FlatList
          ListHeaderComponent={renderHeader()}
          contentInsetAdjustmentBehavior="automatic"
          style={styles.flatlist}
          renderItem={renderItem}
          keyExtractor={(item) => item.id}
          refreshControl={
            <RefreshControl title="Loading..." refreshing={refreshing} onRefresh={onRefresh} />
          }
        />
```

<img width="296" alt="image" src="https://github.com/software-mansion/react-native-screens/assets/688326/8a235a92-975b-4ab8-9505-b022db7f5935">

- When rendering the header directly after app start, things are broken. The header stays in 'big mode' and scrolling doesn't hide it
- When I save any file and react does a hot refresh, the header is fixed and scrolls correctly / hides when scrolling
- When I set `headerLargeTitle` to `false`, then update it in a timeout to `true`, scrolling / hiding / showing works fine, but the header is now small first and I need to scroll to the top to show it in it's big form again
- Delaying rendering of `HomeScreen` (which contains my FlatList) didn't do anything for this issue, neither did delaying only the FlatList 

So far what worked the most reliable with fixing scroll related issues was setting `headerLargeTitle` to false and true again, but as mentioned it puts causes the header to be small first. Gonna do a bit more poking around.

**Comment by dvcrn** [comment]
Man I've figured my issue out, and it was the exact opposite of the hacky workarounds here

It boiled down to 2 issues: 

1) The FlatList was not mounted immediately when the StackView/Screen mounted and only a few ms later. That was because I loaded some data from async storage, so for a fraction of a second I displayed an ActivityIndicator instead of the FlatList (switched with the usual `{isLoading && <loader> || <flatlist}`

2) The FlatList was not the first element within the embedded view. I had something like `{!rows.length && <NoDataView />}` **before** rendering the FlatList. I moved that <NoDataView> after the FlatList and it resolved the issue. 

Making sure the FlatList is **always** mounted when the Screen is mounted, and making sure there are no other elements **before** the FlatList inside the view resolved this issue for me.  I no longer have the app start with a small title, and I have no more issues with scrolling and not hiding the title
